app/scripts/filters/util.js

```diff
@@ -560,4 +560,19 @@ angular.module('openshiftConsole')
         }
       });
     };
+  })
+  // Use of isIE and isEdge is HIGHLY discouraged, only use if absolutely required
+  .filter('isIE', function() {
+    var ua = navigator.userAgent;
+    var isIE = /msie|trident/i.test(ua);
+    return function() {
+      return isIE;
+    };
+  })
+  .filter('isEdge', function() {
+    var ua = navigator.userAgent;
+    var isEdge = /chrome.+? edge/i.test(ua);
+    return function() {
+      return isEdge;
+    };
```

Comment:
I'm fine with this, although we used a constant for iOS:

https://github.com/openshift/origin-web-console/blob/master/app/scripts/app.js#L319